### PR TITLE
APPS-1034 Delete incremental backups

### PR DESCRIPTION
**Pull request by korotkov-aerospike**

### CODE REVIEW

File: pkg/service/s3_context.go [hunk 1]
```diff
@@ -159,26 +159,33 @@ func removeLeadingSlash(s string) string {
 }
 
 // CleanDir cleans the directory with the given name.
-func (s *S3Context) CleanDir(name string) {
-	path := s.Path + "/" + name
+func (s *S3Context) CleanDir(name string) error {
+	path := removeLeadingSlash(s.Path + "/" + name)
```
Comment (korotkov-aerospike) [inline pkg/service/s3_context.go:163]: This line is actually fix

Comment (reugn) [reply pkg/service/s3_context.go:163]: Needs to be tested on both MinIO and S3.

Comment (korotkov-aerospike) [reply pkg/service/s3_context.go:163]: In the ecosystem project

**Comment by korotkov-aerospike** [comment]
I tested on AWS.

File: pkg/service/s3_context_test.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+//go:build !ci
+
+package service
+
+import (
+	"github.com/aerospike/backup/pkg/model"
+	"github.com/aws/smithy-go/ptr"
+	"testing"
```
Comment (reugn) [inline pkg/service/s3_context_test.go:8]: Format imports.

**Review by reugn — APPROVED** [review]

**Pull request merged**